src/main/java/io/opentracing/contrib/activespan/ActiveSpanManager.java

```diff
@@ -0,0 +1,215 @@
+ * and its parents.
+ * <p>
+ * Custom implementations can be provided by:
+ * <ol>
+ * <li>calling {@link #setActiveSpanManager(ActiveSpanManager)} programmatically, or</li>
+ * <li>defining a <code>META-INF/services/io.opentracing.contrib.activespan.ActiveSpanManager</code> service file
+ * containing the classname of the implementation</li>
+ * </ol>
+ *
+ * @author Sjoerd Talsma
+ * @navassoc - activeSpan - io.opentracing.Span
+ */
+public abstract class ActiveSpanManager {
+    private static final Logger LOGGER = Logger.getLogger(ActiveSpanManager.class.getName());
+    private static final AtomicReference<ActiveSpanManager> INSTANCE = new AtomicReference<ActiveSpanManager>();
+
+    /**
+     * Managed object to deactivate an activated {@link Span} with.
+     */
+    public interface SpanDeactivator {
+    }
+
+    /**
+     * Programmatically sets an <code>ActiveSpanManager</code> implementation as <em>the</em> singleton instance.
+     * <p>
+     * Any previous manager is returned so it can be restored if necessary, for example in a testing situation.
+     *
+     * @param instance The overridden implementation to use for in-process span management.
+     * @return Any previous <code>ActiveSpanManager</code> that was initialized before, or <code>null</code>.
+     */
+    public static ActiveSpanManager setActiveSpanManager(ActiveSpanManager instance) {
+        return INSTANCE.getAndSet(instance);
+    }
+
+    /**
+     * Return the active {@link Span}.
+     *
+     * @return The active Span, or the <code>NoopSpan</code> if there is no active span.
+     */
+    public static Span activeSpan() {
+        try {
+            Span activeSpan = getInstance().getActiveSpan();
+            if (activeSpan != null) return activeSpan;
+        } catch (Exception activeSpanException) {
+            LOGGER.log(Level.WARNING, "Could not obtain active span.", activeSpanException);
+        }
+        return NoopSpan.INSTANCE;
+    }
+
+    /**
+     * Makes span the <em>active span</em> within the running process.
+     * <p>
+     * Any exception thrown by the {@link #setActiveSpan(Span) implementation} is logged and will return
+     * no {@link SpanDeactivator} (<code>null</code>) because tracing code must not break application functionality.
+     *
+     * @param span The span to become the active span.
+     * @return The object that will restore any currently <em>active</em> deactivated, or <code>null</code>.
+     * @see #activeSpan()
+     * @see #deactivate(SpanDeactivator)
+     */
+    public static SpanDeactivator activate(Span span) {
+        try {
+            if (span == null) span = NoopSpan.INSTANCE;
+            return getInstance().setActiveSpan(span);
+        } catch (Exception activationException) {
+            LOGGER.log(Level.WARNING, "Could not activate {0}.", new Object[]{span, activationException});
```

Comment:
seems string cat is simpler here. there's questionable value in adding a period vs the cruft. ps did you mean to add the exception as a parameter?